nibabel/gifti/gifti.py

```diff
@@ -298,51 +299,35 @@ def from_array(klass,
         cda.meta = GiftiMetaData.from_dict(meta)
         return cda
 
-    def to_xml(self):
+    def _to_xml_element(self):
         # fix endianness to machine endianness
         self.endian = gifti_endian_codes.code[sys.byteorder]
-        result = ""
-        result += self.to_xml_open()
-        # write metadata
-        if not self.meta is None:
-            result += self.meta.to_xml()
-        # write coord sys
-        if not self.coordsys is None:
-            result += self.coordsys.to_xml()
+
+        data_array = xml.Element('DataArray', attrib={
+            'Intent': intent_codes.niistring[self.intent],
+            'DataType': data_type_codes.niistring[self.datatype],
+            'ArrayIndexingOrder': array_index_order_codes.label[self.ind_ord],
+            'Dimensionality': str(self.num_dim),
+            'Encoding': gifti_encoding_codes.specs[self.encoding],
+            'Endian': gifti_endian_codes.specs[self.endian],
+            'ExternalFileName': self.ext_fname,
+            'ExternalFileOffset': self.ext_offset})
+        for di, dn in enumerate(self.dims):
+            data_array.attrib['Dim%d' % di] = str(dn)
+
+        if self.meta is not None:
+            data_array.append(self.meta._to_xml_element())
+        if self.coordsys is not None:
+            data_array.append(self.coordsys._to_xml_element())
         # write data array depending on the encoding
         dt_kind = data_type_codes.dtype[self.datatype].kind
-        result += data_tag(self.data,
-                           gifti_encoding_codes.specs[self.encoding],
-                           KIND2FMT[dt_kind],
-                           self.ind_ord)
-        result = result + self.to_xml_close()
-        return result
-
-    def to_xml_open(self):
-        out = """<DataArray Intent="%s"
-\tDataType="%s"
-\tArrayIndexingOrder="%s"
-\tDimensionality="%s"
-%s\tEncoding="%s"
-\tEndian="%s"
-\tExternalFileName="%s"
-\tExternalFileOffset="%s">\n"""
-        di = ""
-        for i, n in enumerate(self.dims):
-            di = di + '\tDim%s=\"%s\"\n' % (str(i), str(n))
-        return out % (intent_codes.niistring[self.intent],
-                      data_type_codes.niistring[self.datatype],
-                      array_index_order_codes.label[self.ind_ord],
-                      str(self.num_dim),
-                      str(di),
-                      gifti_encoding_codes.specs[self.encoding],
-                      gifti_endian_codes.specs[self.endian],
-                      self.ext_fname,
-                      self.ext_offset,
-                      )
-
-    def to_xml_close(self):
-        return "</DataArray>\n"
```

Comment:
Stupid, but do you think we should have deprecated `to_xml_open` and `to_xml_close`? They were public, even if they were kind of weird. If so, I think the easiest way is probably just to restore them verbatim, since neither corresponds to a full XML element.